### DISCUSSION TITLE
Can I change rediness probe condition for kafka connect?

### DISCUSSION THREAD

**Discussion by anvaari** [strimzi/strimzi-kafka-operator#10027] [Q&A] [answered]
I need to change the readiness for kafka connect pods from checking the healthiness of the rest endpoint to counting the number of loaded connectors (it takes about 5-10 minutes after the healthiness of the rest endpoint). Is it possible? Because I couldn't find any option [here](https://strimzi.io/docs/operators/latest/configuring.html#type-Probe-reference)

**Comment by scholzj** [comment, marked as answer]
No, you cannot change what the readiness probe does. You can change only the timing of it.

**Reply by anvaari** [reply]
Thanks. Do you think I can create a feature request for it? It is needed; otherwise, the cluster will not be reliable.

**Reply by scholzj** [reply]
I don't think so. The operator needs to understand the readiness probe semantics and it needs to fit how it is designed. I'm not sure what exactly `number of loaded connectors` means, but it does not seem to fit the semantics of the readiness probe. Connector plugins are loaded at startup. Connector instances are part of the whole cluster and not a single Pod, so they have no relationship to Connect readiness either.

**Reply by anvaari** [reply]
Oh you are right, number of loaded connector is cluster wide and can't be condition for a single pod readiness.
But I think having the option to change the condition is good in some cases. User can add bah file in image, and use that for readiness condition.

**Reply by anvaari** [reply]
Suppoae I have three replicas for kafka connect, if the readiness for pod is set to healthiness of REST endpoint, we will have downtime for each rolling upgrade, because it take about 5-10 minutes for connectors to be load. So the pod become ready, old pod terminated, but no data come until all connectors loaded

**Reply by scholzj** [reply]
>Oh you are right, number of loaded connector is cluster wide and can't be condition for a single pod readiness.
But I think having the option to change the condition is good in some cases. User can add bah file in image, and use that for readiness condition.

Honestly, I so far did not hear an idea how to do the readiness differently that I would consider reasonable But also as I said, the operator needs to know what it does and how it works because it is relying on it.

>Suppoae I have three replicas for kafka connect, if the readiness for pod is set to healthiness of REST endpoint, we will have downtime for each rolling upgrade, because it take about 5-10 minutes for connectors to be load. So the pod become ready, old pod terminated, but no data come until all connectors loaded

I still don't understand what does it really mean when you say that it takes 5-10 minutes for a conenctor to get loaded. I'm not aware of any such thing in Kafka Connect.

**Reply by anvaari** [reply]
I don't know, mybe it's my problem. Not general problem. Thanks, I'll investigate more

**Reply by anvaari** [reply]
The problem related to the kafka. It cause cold start of kafka connect. It's not related to Strimzi of kafka connect
